File: 50-Models/Elysio.Models/DTOs/RoomDTO.cs
namespace Elysio.Models.DTOs;

public record RoomDTO
{
    public Guid Id { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public List<AgentDTO> Agents { get; set; } = [];
    public List<ConversationDTO> Conversations { get; set; } = [];
}

File: 10-Client/elysio.crew.front/src/models/interfaces/RoomDTO.ts
import type { AgentDTO } from "../interfaces/AgentDTO";
import type { ConversationDTO } from "../interfaces/ConversationDTO";

export type RoomDTO = {
  id: string;
  createdAt: string;
  updatedAt: string;
  name: string;
  description?: string;
  agents: AgentDTO[];
  conversations: ConversationDTO[];
};
